k2/python/k2/dense_fsa.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+
+import _k2
+import numpy as np
+import torch
+
+
+def dense_fsa(log_probs: torch.Tensor,
+              input_lengths: torch.Tensor) -> _k2.DenseFsaVec:
+    '''Construct a DenseFsa from neural net log-softmax outputs.
+
+    Args:
+      log_probs:
+        A 3-D tensor of dtype ``torch.float32`` with shape ``(N, T, C)``, where
+        ``N`` is the batch size, ``T`` the maximum input length, and ``C`` the
+        number of output classes.
+      input_lengths:
+        A 1-D tensor of dtype ``torch.int32`` with size ``(N)``, where ``N`` is
+        the batch size. It represents the length of the input (must each NOT
```

Comment:
each must?

Reply:
I think both are fine.

this is from
https://pytorch.org/docs/stable/generated/torch.nn.CTCLoss.html

> Input_lengths: Tuple or tensor of size (N)(N) , where N = \text{batch size}N=batch size . It represent the lengths of the inputs (must each be \leq T≤T ).

Reply:
OK, just feel a little hard for me (a non-native) to read.

Reply:
Comments are updated.